mongodb/src/main/java/org/apache/zeppelin/mongodb/MongoDbInterpreter.java

```diff
@@ -0,0 +1,190 @@
+    try {
+      FileUtils.write(scriptFile,
+          SHELL_EXTENSION
+          .replace("__ZEPPELIN_TABLE_LIMIT__", getProperty("mongo.shell.command.table.limit")) +
+          script);
+    }
+    catch (IOException e) {
+      LOGGER.error("Can not write script in temp file", e);
+      return new InterpreterResult(Code.ERROR, e.getMessage());
+    }
+
+    InterpreterResult result = new InterpreterResult(InterpreterResult.Code.SUCCESS);
+
+    final DefaultExecutor executor = new DefaultExecutor();
+    final ByteArrayOutputStream errorStream = new ByteArrayOutputStream();
+    executor.setStreamHandler(new PumpStreamHandler(context.out, errorStream));
+    executor.setWatchdog(new ExecuteWatchdog(commandTimeout)); //ExecuteWatchdog.INFINITE_TIMEOUT
+
+    final CommandLine cmdLine = CommandLine.parse(getProperty("mongo.shell.path"));
+    cmdLine.addArgument("--quiet", false);
+
+    if (!StringUtils.isEmpty(getProperty("mongo.server.username"))) {
+      cmdLine.addArgument("-u", false);
+      cmdLine.addArgument(getProperty("mongo.server.username"), false);
+      cmdLine.addArgument("-p", false);
+      cmdLine.addArgument(getProperty("mongo.server.password"), false);
+      
+      if (!StringUtils.isEmpty(getProperty("mongo.server.authentdatabase"))) {
+        cmdLine.addArgument("--authenticationDatabase", false);
+        cmdLine.addArgument(getProperty("mongo.server.authentdatabase"), false);
+      }
+    }
+
+    cmdLine.addArgument(dbAddress, false);
+    cmdLine.addArgument(scriptFile.getAbsolutePath(), false);
+
+    try {
+      executor.execute(cmdLine);
+      runningProcesses.put(context.getParagraphId(), executor);
+    }
+    catch (ExecuteException e) {
+      LOGGER.error("Can not run script in paragraph {}", context.getParagraphId(), e);
+
+      final int exitValue = e.getExitValue();
+      Code code = Code.ERROR;
+      String msg = errorStream.toString();
+      if (exitValue == 143) {
+        code = Code.INCOMPLETE;
+        msg = msg + "Paragraph received a SIGTERM.\n";
+        LOGGER.info("The paragraph {} stopped executing: {}", context.getParagraphId(), msg);
+      }
+      msg += "ExitValue: " + exitValue;
+      result = new InterpreterResult(code, msg);
+    }
+    catch (IOException e) {
+      LOGGER.error("Can not run script in paragraph {}", context.getParagraphId(), e);
+      result = new InterpreterResult(Code.ERROR, e.getMessage());
+    }
+    finally {
+      FileUtils.deleteQuietly(scriptFile);
+    }
+
+    return result;
+  }
+
+  @Override
+  public int getProgress(InterpreterContext context) {
+    return 0;
+  }
+
+  @Override
+  public void cancel(InterpreterContext context) {
+    stopProcess(context.getParagraphId());
+    FileUtils.deleteQuietly(new File(getScriptFileName(context.getParagraphId())));
+  }
+
+  @Override
+  public Scheduler getScheduler() {
+    return SchedulerFactory.singleton().createOrGetParallelScheduler("mongo", 10);
```

Comment:
10 is hard coded, should it be configurable ?

Reply:
mongo.interpreter.concurrency.max  property supplied